spec/uber_task/uber_task_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+describe UberTask do
+  describe '.on_report' do
+    context 'when no block is passed' do
+      it 'returns nil' do
+        described_class.run do |task|
+          expect(described_class.on_report).to be_nil
+        end
+      end
+    end
+
+    context 'when a block is passed' do
+      it 'returns nil' do
+        described_class.run do |task|
+          result = described_class.on_report do
+            "Added some logs"
+          end
+
+          expect(result.call).to eq("Added some logs")
```

Comment:
We need to update the `on_report` to always return nil, the returned value was not intentional.

we need to check that the block was added to the task's handlers (you can see [here ](https://github.com/shakacode/uber_task/blob/main/lib/uber_task.rb#L29C21-L29C30)that tasks are expected to have a handlers attribute).